sdk/src/main/java/com/horizen/account/transaction/EthereumTransaction.java

```diff
@@ -349,22 +346,30 @@ public String toString() {
                 this.getTo() != null ? this.getTo() : "0x",
                 Numeric.toHexStringWithPrefix(this.getValue()),
                 this.getData() != null ? Numeric.toHexString(this.getData()) : "",
-                isSigned() ? getSignature().toString() : ""
+                isSigned() ? new SignatureSecp256k1(getSignatureData()).toString() : ""
         );
     }
 
-    /*
-     * from: "0xEB014f8c8B418Db6b45774c326A0E64C78914dC0",
-     * gasPrice: "20000000000",
-     * gas: "21000",
-     * to: '0x3535353535353535353535353535353535353535',
-     * value: "1000000000000000000",
-     * data: ""
-     */
     @Override
     public byte[] messageToSign() {
-        if (this.transaction.getType().isLegacy() && this.isSigned())
+        if (this.transaction.getType().isLegacy() && this.isSigned()) {
+            // the chainid might be set also in legacy case due to EIP155
             return ((SignedRawTransaction) this.transaction).getEncodedTransaction(this.getChainId());
+        }
         return TransactionEncoder.encode(this.transaction);
     }
+
+    // Util function
+    // w3j private method in TransactionEncoder, it returns a byte array with Long.BYTES length
+    public static byte[] convertToBytes(long x) {
+        ByteBuffer buffer = ByteBuffer.allocate(Long.BYTES);
+        buffer.putLong(x);
+        return buffer.array();
+    }
+
+    // w3j way for converting bytes, it works also with generic byte contents (not only Long.BYTES byte arrays)
+    public static long convertToLong(byte[] bytes) {
+        BigInteger bi = Numeric.toBigInt(bytes);
+        return bi.longValue();
+    }
```

Comment:
I would suggest to move it a separate utility class into `com.horizen.account.utils` package.

Reply:
Ok, moved in a new EthereumTransactionUtils class in the package